File: dto-mapping-benchmark/src/main/java/com/example/dtomappingbenchmark/v3/service/UserService.java
package com.example.dtomappingbenchmark.v3.service;

import com.example.dtomappingbenchmark.v3.entity.User;
import com.example.dtomappingbenchmark.v3.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;


@Service
@RequiredArgsConstructor
public class UserService {
    private final UserRepository userRepository;

    public User create(User user) {
        return userRepository.save(user);
    }

    public User get(Long id) {
        return userRepository.findById(id).orElseThrow();
    }
}

File: object-mapping-benchmark-v1/src/main/kotlin/yu/selab/cj/objectmappingbenchmarkv1/user/service/UserService.kt
package yu.selab.cj.objectmappingbenchmarkv1.user.service

import org.springframework.stereotype.Service
import yu.selab.cj.objectmappingbenchmarkv1.user.command.UserDto
import yu.selab.cj.objectmappingbenchmarkv1.user.entity.User
import yu.selab.cj.objectmappingbenchmarkv1.user.repository.UserRepository

@Service
class UserService(
    private val userRepository : UserRepository
) {

    fun getUserList() = userRepository.findAll()

    fun getUser(id : Long) = userRepository.findById(id).orElseThrow {
        IllegalArgumentException("user not found")
    }

    fun createUser(dto : UserDto){
        val user = User(dto.name, dto.password)
        userRepository.save(user)
    }

    fun deleteUser(id : Long)
        = userRepository.deleteById(id)

}